### PR TITLE
Add Database kafka schema registry

### PR DESCRIPTION
**Pull request by BellionBastien**
# Description

Adding resource and datasource  for newly implemented schema registry ACL on kafka databases.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A: `make testacc TESTARGS="-run TestAccCloudProjectDatabaseKafkaSchemaregistryacl_basic"`
- [x] Test B: `make testacc TESTARGS="-run TestAccCloudProjectDatabaseKafkaSchemaregistryaclsDataSource_basic"`

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings or issues
- [x] I have added acceptance tests that prove my fix is effective or that my feature works
- [x] New and existing acceptance tests pass locally with my changes
- [x] I ran succesfully `go mod vendor` if I added or modify `go.mod` file

### CODE REVIEW

**Review by lpatte — APPROVED** [review]

File: website/docs/r/cloud_project_database_kafka_schemaregistryacl.html.markdown [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+subcategory : "Managed Databases"
+---
+
+# ovh_cloud_project_database_kafka_schemaregistryacl
+
+Creates a schema registry ACL for a kafka cluster associated with a public cloud project.
```
Comment (scraly) [inline website/docs/r/cloud_project_database_kafka_schemaregistryacl.html.markdown:7]:
```suggestion
Creates a schema registry ACL for a Kafka cluster associated with a public cloud project.
```

File: website/docs/r/cloud_project_database_kafka_schemaregistryacl.html.markdown [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    delete = "45m"
+  }
+}
+```
+* `create` - (Default 20m)
+* `delete` - (Default 20m)
+
+## Import
+
+OVHcloud Managed kafka clusters schema registry ACLs can be imported using the `service_name`, `cluster_id` and `id` of the schema registry acl, separated by "/" E.g.,
```
Comment (scraly) [inline website/docs/r/cloud_project_database_kafka_schemaregistryacl.html.markdown:73]:
```suggestion
OVHcloud Managed Kafka clusters schema registry ACLs can be imported using the `service_name`, `cluster_id` and `id` of the schema registry ACL, separated by "/" E.g.,
```

File: ovh/data_cloud_project_database_kafka_schemaregistryacl.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+			// Computed
+			"permission": {
+				Type:        schema.TypeString,
+				Description: "Permission to give to this username on this resource",
+				Computed:    true,
+			},
+			"resource": {
+				Type:        schema.TypeString,
+				Description: "Resource affected by this acl",
```
Comment (scraly) [inline ovh/data_cloud_project_database_kafka_schemaregistryacl.go:42]:
```suggestion
				Description: "Resource affected by this ACL",
```

File: ovh/data_cloud_project_database_kafka_schemaregistryacl.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+				Computed:    true,
+			},
+			"username": {
+				Type:        schema.TypeString,
+				Description: "Username affected by this acl",
```
Comment (scraly) [inline ovh/data_cloud_project_database_kafka_schemaregistryacl.go:47]:
```suggestion
				Description: "Username affected by this ACL",
```

File: ovh/data_cloud_project_database_kafka_schemaregistryacl.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+	for k, v := range res.ToMap() {
+		if k != "id" {
+			d.Set(k, v)
+		} else {
+			d.SetId(fmt.Sprint(v))
+		}
+	}
+
+	log.Printf("[DEBUG] Read acl %+v", res)
```
Comment (scraly) [inline ovh/data_cloud_project_database_kafka_schemaregistryacl.go:80]:
```suggestion
	log.Printf("[DEBUG] Read ACL %+v", res)
```

File: ovh/data_cloud_project_database_kafka_schemaregistryacls.go [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+	serviceName := d.Get("service_name").(string)
+	clusterId := d.Get("cluster_id").(string)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/kafka/%s/schemaRegistryAcl",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterId),
+	)
+	res := make([]string, 0)
+
+	log.Printf("[DEBUG] Will read schema registry acls from cluster %s from project %s", clusterId, serviceName)
```
Comment (scraly) [inline ovh/data_cloud_project_database_kafka_schemaregistryacls.go:53]:
```suggestion
	log.Printf("[DEBUG] Will read schema registry ACLs from cluster %s from project %s", clusterId, serviceName)
```

File: ovh/data_cloud_project_database_kafka_schemaregistryacls.go [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+			"cluster_id": {
+				Type:        schema.TypeString,
+				Description: "Id of the database cluster",
+				Required:    true,
+			},
+
+			// Computed
+			"acl_ids": {
+				Type:        schema.TypeList,
+				Description: "List of schema registry acl ids",
```
Comment (scraly) [inline ovh/data_cloud_project_database_kafka_schemaregistryacls.go:34]:
```suggestion
				Description: "List of schema registry ACL IDs",
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 8]
```diff
@@ -0,0 +1,173 @@
...
+			},
+			"permission": {
+				Type:        schema.TypeString,
+				Description: "Permission to give to this username on this resource",
+				ForceNew:    true,
+				Required:    true,
+			},
+			"resource": {
+				Type:        schema.TypeString,
+				Description: "Resource affected by this acl",
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:52]:
```suggestion
				Description: "Resource affected by this ACL",
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 9]
```diff
@@ -0,0 +1,173 @@
...
+				ForceNew:    true,
+				Required:    true,
+			},
+			"username": {
+				Type:        schema.TypeString,
+				Description: "Username affected by this acl",
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:58]:
```suggestion
				Description: "Username affected by this ACL",
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 10]
```diff
@@ -0,0 +1,173 @@
...
+	clusterId := d.Get("cluster_id").(string)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/kafka/%s/schemaRegistryAcl",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterId),
+	)
+	params := (&CloudProjectDatabaseKafkaSchemaregistryaclCreateOpts{}).FromResource(d)
+	res := &CloudProjectDatabaseKafkaSchemaregistryaclResponse{}
+
+	log.Printf("[DEBUG] Will create schema registry acl: %+v for cluster %s from project %s", params, clusterId, serviceName)
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:97]:
```suggestion
	log.Printf("[DEBUG] Will create schema registry ACL: %+v for cluster %s from project %s", params, clusterId, serviceName)
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+	err := config.OVHClient.Post(endpoint, params, res)
+	if err != nil {
+		return diag.Errorf("calling Post %s with params %+v:\n\t %q", endpoint, params, err)
+	}
+
+	log.Printf("[DEBUG] Waiting for schema registry acl %s to be READY", res.Id)
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:103]:
```suggestion
	log.Printf("[DEBUG] Waiting for schema registry ACL %s to be READY", res.Id)
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+	err = waitForCloudProjectDatabaseKafkaSchemaregistryaclReady(ctx, config.OVHClient, serviceName, clusterId, res.Id, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Errorf("timeout while waiting schema registry acl %s to be READY: %s", res.Id, err.Error())
+	}
+	log.Printf("[DEBUG] schema registry acl %s is READY", res.Id)
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:108]:
```suggestion
	log.Printf("[DEBUG] schema registry ACL %s is READY", res.Id)
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+	id := d.Id()
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/kafka/%s/schemaRegistryAcl/%s",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterId),
+		url.PathEscape(id),
+	)
+	res := &CloudProjectDatabaseKafkaSchemaregistryaclResponse{}
+
+	log.Printf("[DEBUG] Will read schema registry acl %s from cluster %s from project %s", id, clusterId, serviceName)
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:128]:
```suggestion
	log.Printf("[DEBUG] Will read schema registry ACL %s from cluster %s from project %s", id, clusterId, serviceName)
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+	clusterId := d.Get("cluster_id").(string)
+	id := d.Id()
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/kafka/%s/schemaRegistryAcl/%s",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterId),
+		url.PathEscape(id),
+	)
+
+	log.Printf("[DEBUG] Will delete schema registry acl  %s from cluster %s from project %s", id, clusterId, serviceName)
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:157]:
```suggestion
	log.Printf("[DEBUG] Will delete schema registry ACL %s from cluster %s from project %s", id, clusterId, serviceName)
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+	err := config.OVHClient.Delete(endpoint, nil)
+	if err != nil {
+		return diag.FromErr(helpers.CheckDeleted(d, err, endpoint))
+	}
+
+	log.Printf("[DEBUG] Waiting for schema registry acl %s to be DELETED", id)
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:163]:
```suggestion
	log.Printf("[DEBUG] Waiting for schema registry ACL %s to be DELETED", id)
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+	err = waitForCloudProjectDatabaseKafkaSchemaregistryaclDeleted(ctx, config.OVHClient, serviceName, clusterId, id, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return diag.Errorf("timeout while waiting schema registry acl %s to be DELETED: %s", id, err.Error())
+	}
+	log.Printf("[DEBUG] schema registry acl %s is DELETED", id)
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:168]:
```suggestion
	log.Printf("[DEBUG] schema registry ACL %s is DELETED", id)
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+	log.Printf("[DEBUG] Will delete schema registry acl  %s from cluster %s from project %s", id, clusterId, serviceName)
+	err := config.OVHClient.Delete(endpoint, nil)
+	if err != nil {
+		return diag.FromErr(helpers.CheckDeleted(d, err, endpoint))
+	}
+
+	log.Printf("[DEBUG] Waiting for schema registry acl %s to be DELETED", id)
+	err = waitForCloudProjectDatabaseKafkaSchemaregistryaclDeleted(ctx, config.OVHClient, serviceName, clusterId, id, d.Timeout(schema.TimeoutDelete))
+	if err != nil {
+		return diag.Errorf("timeout while waiting schema registry acl %s to be DELETED: %s", id, err.Error())
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:166]:
```suggestion
		return diag.Errorf("timeout while waiting schema registry ACL %s to be DELETED: %s", id, err.Error())
```

File: ovh/data_cloud_project_database_kafka_schemaregistryacl.go [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+				DefaultFunc: schema.EnvDefaultFunc("OVH_CLOUD_PROJECT_SERVICE", nil),
+			},
+			"cluster_id": {
+				Type:        schema.TypeString,
+				Description: "Id of the database cluster",
+				Required:    true,
+			},
+			"id": {
+				Type:        schema.TypeString,
+				Description: "Shema registry acl ID",
```
Comment (scraly) [inline ovh/data_cloud_project_database_kafka_schemaregistryacl.go:30]:
```suggestion
				Description: "Shema registry ACL ID",
```

File: ovh/data_cloud_project_database_kafka_schemaregistryacl.go [hunk 19]
```diff
@@ -0,0 +1,82 @@
...
+	id := d.Get("id").(string)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/kafka/%s/schemaRegistryAcl/%s",
+		url.PathEscape(serviceName),
+		url.PathEscape(clusterID),
+		url.PathEscape(id),
+	)
+	res := &CloudProjectDatabaseKafkaAclResponse{}
+
+	log.Printf("[DEBUG] Will read schema registry acl %s from cluster %s from project %s", id, clusterID, serviceName)
```
Comment (scraly) [inline ovh/data_cloud_project_database_kafka_schemaregistryacl.go:67]:
```suggestion
	log.Printf("[DEBUG] Will read schema registry ACL %s from cluster %s from project %s", id, clusterID, serviceName)
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 20]
```diff
@@ -0,0 +1,173 @@
...
+	log.Printf("[DEBUG] Will create schema registry acl: %+v for cluster %s from project %s", params, clusterId, serviceName)
+	err := config.OVHClient.Post(endpoint, params, res)
+	if err != nil {
+		return diag.Errorf("calling Post %s with params %+v:\n\t %q", endpoint, params, err)
+	}
+
+	log.Printf("[DEBUG] Waiting for schema registry acl %s to be READY", res.Id)
+	err = waitForCloudProjectDatabaseKafkaSchemaregistryaclReady(ctx, config.OVHClient, serviceName, clusterId, res.Id, d.Timeout(schema.TimeoutCreate))
+	if err != nil {
+		return diag.Errorf("timeout while waiting schema registry acl %s to be READY: %s", res.Id, err.Error())
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:106]:
```suggestion
		return diag.Errorf("timeout while waiting schema registry ACL %s to be READY: %s", res.Id, err.Error())
```

File: ovh/resource_cloud_project_database_kafka_schemaregistryacl.go [hunk 21]
```diff
@@ -0,0 +1,173 @@
...
+
+	for k, v := range res.ToMap() {
+		if k != "id" {
+			d.Set(k, v)
+		} else {
+			d.SetId(fmt.Sprint(v))
+		}
+	}
+
+	log.Printf("[DEBUG] Read schema registry acl %+v", res)
```
Comment (scraly) [inline ovh/resource_cloud_project_database_kafka_schemaregistryacl.go:141]:
```suggestion
	log.Printf("[DEBUG] Read schema registry ACL %+v", res)
```

**Review by scraly — CHANGES_REQUESTED** [review]
The new feature is working, thanks! :-) 

Just some remarks, typos and few changes requested.

**Pull request merged**